diff --git a/theme/views_slideshow.theme.inc b/theme/views_slideshow.theme.inc
index 81fcd69..c9b70e1 100644
--- a/theme/views_slideshow.theme.inc
+++ b/theme/views_slideshow.theme.inc
@@ -70,10 +70,22 @@ function _views_slideshow_preprocess_views_slideshow(&$vars) {
     foreach ($addons as $addon_id => $addon_info) {
       foreach ($addon_info['accepts'] as $imp_key => $imp_value) {
         if (is_array($imp_value)) {
-          $methods[$imp_key][] = preg_replace('/_(.?)/e',"strtoupper('$1')", $addon_id);
+          $methods[$imp_key][] = preg_replace_callback(   // php5.x preg_replace_callback() method
+            '/_(.?)/',
+            function ($m) {
+              return strtoupper($m[1]);
+            },
+            $addon_id
+          );
         }
         else {
-          $methods[$imp_value][] = preg_replace('/_(.?)/e',"strtoupper('$1')", $addon_id);
+          $methods[$imp_value][] = preg_replace_callback(  // php5.x preg_replace_callback() method
+            '/_(.?)/',
+            function ($m) {
+              return strtoupper($m[1]);
+            },
+            $addon_id
+          );
         }
       }
     }
@@ -184,10 +196,15 @@ function theme_views_slideshow_pager_widget_render($vars) {
     'viewsSlideshowPager' => array(
       $vars['vss_id'] => array(
         $vars['location'] => array(
-          'type' => preg_replace('/_(.?)/e',"strtoupper('$1')", $vars['settings']['type']),
+          'type' => preg_replace_callback(   // php5.x preg_replace_callback() method
+            '/_(.?)/',
+            function ($m) {
+              return strtoupper($m[1]);
+            },
+            $vars['settings']['type']),
+          ),
         ),
       ),
-    ),
   );
 
   drupal_add_js($js_vars, 'setting');
@@ -272,10 +289,15 @@ function theme_views_slideshow_controls_widget_render($vars) {
     'viewsSlideshowControls' => array(
       $vars['vss_id'] => array(
         $vars['location'] => array(
-          'type' => preg_replace('/_(.?)/e',"strtoupper('$1')", $vars['settings']['type']),
+          'type' => preg_replace_callback(   // php5.x preg_replace_callback() method
+            '/_(.?)/',
+            function ($m) {
+              return strtoupper($m[1]);
+            },
+            $vars['settings']['type']),
+          ),
         ),
       ),
-    ),
   );
 
   drupal_add_js($js_vars, 'setting');
